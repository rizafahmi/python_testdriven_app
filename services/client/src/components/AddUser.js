import React from 'react'

const AddUser = props => {
  return (
    <form onSubmit={props.handleAddUser}>
      <div className='form-group'>
        <input
          name='username'
          className='form-control input-lg'
          type='text'
          placeholder='Enter a username'
          required
          value={props.username}
          onChange={props.handleInputChange}
        />
      </div>
      <div className='form-group'>
        <input
          name='email'
          className='form-control input-lg'
          type='email'
          placeholder='Enter your email'
          required
          value={props.email}
          onChange={props.handleInputChange}
        />
      </div>
      <input
        type='submit'
        className='btn btn-warning btn-lg btn-block'
        value='Submit'
      />
    </form>
  )
}

export default AddUser
