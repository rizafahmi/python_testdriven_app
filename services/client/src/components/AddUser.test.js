import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import AddUser from './AddUser'

test('AddUser renders properly', () => {
  const wrapper = shallow(<AddUser />)
  const element = wrapper.find('form')
  const input = element.find('input')
  expect(input.length).toBe(3)
  expect(input.get(0).props.name).toBe('username')
  expect(input.get(1).props.name).toBe('email')
  expect(input.get(2).props.type).toBe('submit')
})
test('AddUser renders a snapshot properly', () => {
  const tree = renderer.create(<AddUser />).toJSON()
  expect(tree).toMatchSnapshot()
})
