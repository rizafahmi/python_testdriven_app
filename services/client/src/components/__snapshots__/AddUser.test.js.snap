// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser renders a snapshot properly 1`] = `
<form
  onSubmit={undefined}
>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="username"
      onChange={undefined}
      placeholder="Enter a username"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="email"
      onChange={undefined}
      placeholder="Enter your email"
      required={true}
      type="email"
      value={undefined}
    />
  </div>
  <input
    className="btn btn-warning btn-lg btn-block"
    type="submit"
    value="Submit"
  />
</form>
`;
